scripts/BindingGenerator.lua

```diff
@@ -0,0 +1,446 @@
+		local k = priorityList[i];
+		local proc = procTable[str][k]
+		if proc then
+			if not typeDefGenerated then
+				typedefs:Add( GetProcedureTypedef( proc ) );
+				typeDefGenerated = true;
+			end
+			local vendor = k == "core" and "" or k;
+			loader:AddFormat(
+				'\tif (!%s) %s = (%s)cmgr->getProcAddress("%s");\n',
+				str, str, pfnFormat( proc.name:upper() ), ("gl%s%s")(str,vendor)
+			);
+		end
+	end
+end
+
+
+------------ Write files ------------
+
+-- Write loader header
+local f = io.open( sourceTreePath .. "/include/mt_opengl.h", "wb" );
+f:write[[
+// This code was generated by scripts/BindingGenerator.lua
+// Do not modify it, modify and run the generator instead.
+
+#pragma once
+
+#include <string>
+#include <unordered_set>
+#include "irrTypes.h"
+#include "IContextManager.h"
+#include <KHR/khrplatform.h>
+
+#ifndef APIENTRY
+	#define APIENTRY
+#endif
+#ifndef APIENTRYP
+	#define APIENTRYP APIENTRY *
+#endif
+#ifndef GLAPI
+	#define GLAPI extern
+#endif
+
+]];
+
+f:write[[
+class OpenGLProcedures {
+private:
+]];
+f:write( definitions:Concat( "\n" ) );
+f:write( "\n" );
+f:write[[
+	// The script will miss this particular typedef thinking it's a PFN,
+	// so we have to paste it in manually. It's the only such type in OpenGL.
+	typedef void (APIENTRY *GLDEBUGPROC)
+		(GLenum source,GLenum type,GLuint id,GLenum severity,GLsizei length,const GLchar *message,const void *userParam);
+
+]]
+f:write( typedefs:Concat( "\n" ) );
+f:write( "\n\n" );
+f:write [[
+	std::unordered_set<std::string> extensions;
+	void* GetProcAddress();
+public:
+	// Call this once after creating the context.
+	void LoadAllProcedures(irr::video::IContextManager *cmgr);
+	// Check if an extension is supported.
+	inline bool IsExtensionPresent(const std::string &ext)
+	{
+		return extensions.find(ext) != extensions.end();
+	}
+
+]];
+f:write( pointers:Concat( "\n" ) );
+f:write( "\n\n" );
+f:write( cppConsts:Concat( "\n" ) );
+f:write( "\n\n" );
+f:write( "};\n" );
+f:write( "\n//Global GL procedures object.\n" );
+f:write( [[
+#ifdef WIN32
+	#ifdef MINETEST_CLIENT
+		#define DLL_SHARED __declspec(dllimport)
+	#else
+		#define DLL_SHARED __declspec(dllexport)
+	#endif
+#else
+	// On Linux, this supposedly works out of the box.
+	#define DLL_SHARED
+	// And on OSX, everything is linked statically anyway.
+#endif
+
+]] );
+f:write( "DLL_SHARED extern OpenGLProcedures GL;\n" );
+f:close();
+
+-- Write loader implementation
+f = io.open( sourceTreePath .. "/source/Irrlicht/mt_opengl_loader.cpp", "wb" );
+f:write[[
+#include "mt_opengl.h"
+#include <string>
+#include <sstream>
+
+OpenGLProcedures GL = OpenGLProcedures();
+
+void OpenGLProcedures::LoadAllProcedures(irr::video::IContextManager *cmgr)
+{
+
+]];
+f:write( loader:Concat() );
+f:write[[
+
+	// get the extension string, chop it up
+	std::string ext_string = std::string((char*)GetString(EXTENSIONS));
+	std::stringstream ext_ss(ext_string);
+	std::string tmp;
+	while (getline(ext_ss, tmp, ' '))
```

Comment:
I don't see an `use namespace` so how does this even work?
```suggestion
	while (std::getline(ext_ss, tmp, ' '))
```

Reply:
I have no idea, but it doesn't hurt to add.